src/authentication/abax-auth.ts

```diff
@@ -207,6 +218,14 @@ export class AbaxAuth {
       return undefined;
     }
 
+    // check if token will expire within the minimum lifetime
+    if (
+      this.credentials.expiresAt.getTime() - Date.now() <
+      this.minimumTokenLifetime * 1000
+    ) {
+      await this.refreshCredentials();
+    }
+
     // check if token is expired
     if (
       this.credentials.refreshToken &&
```

Comment:
I think the extra step where we check if the token is expired is unnecessary here, so just remove it.